### PR TITLE
Allow us to set which end of the robot is considered the front

### PR DESCRIPTION
**Pull request by markmellors**
This allows us to drive "backwards" (steereable wheels at the back) for the Ecodisaster challenge, without any other changes.

### CODE REVIEW

**Comment by markmellors** [comment]
I think its worth thinking about if I've made this flip in the best place. I was hoping it would only require a change in the navigator, then everything onwards in the chain would be the same. So, as coded in this PR, a request to go forward is received in the navigator, it could flip that to reverse if we're considering the back as the front, then the statemanager makes the robot to go in reverse. Unfortunately odometry (state_estimator) also needs to know the direction, either by offsetting the heading 180 degrees somewhere, or in how the X & Y estimates are updated. 
I think theres probably an alternative implementation where the navigator is the same for which ever direction we're going, and its the state_manager that changes. I'm not sure what would happen in that case with state_esitmator. maybe it would still need to change. 
I did it this way in case there's a dynamics difference between chassis forwards and chassis backwards, but I haven't thought the options through in that much detail.

File: libs/state_estimator/src/state_estimator.cpp [hunk 1]
```diff
@@ -46,6 +46,11 @@ namespace STATE_ESTIMATOR {
                 sleep_ms(1000);
             }
         }
+        if (direction == CONFIG::DrivingDirection::SteerableWheelsAtFront){
```
Comment (robberwick) [inline libs/state_estimator/src/state_estimator.cpp:49]: again, we could just use the `DriveDirection` enum alue directly here, if we set the values (see above)

**Review by robberwick — CHANGES_REQUESTED** [review]
Looks ok - I think i'd simplify the driving direction enum thiing a little (see comments).

In terms of broader implementation approaches, this seems fine to me - it's certainly good enough to go with until we encounter any issues :+1:.

File: libs/config/include/drivetrain_config.h [hunk 2]
```diff
@@ -57,21 +63,26 @@ namespace CONFIG {
 
     // Define WHEEL_DIAMETER and GEAR_RATIO based on CURRENT_CHALLENGE
     #if (CURRENT_CHALLENGE == ECO_DISASTER)
-        constexpr float WHEEL_DIAMETER = LARGE_WHEEL_DIAMETER;
-        constexpr float GEAR_RATIO = 42.0 / 18.0 * GEARMOTOR_RATIO;
+        constexpr float WHEEL_DIAMETER = SMALL_WHEEL_DIAMETER; //LARGE_WHEEL_DIAMETER;
```
[the remark this answers is not in the document]
Comment (markmellors) [reply libs/config/include/drivetrain_config.h:66]: That would be less verbose, but I think I prefer the exhaustive list approach, it makes it clearer to me in this case. I think we'd still need the If..elif..else as there are that many different configurations.

File: libs/config/include/drivetrain_config.h [hunk 3]
```diff
@@ -24,6 +24,12 @@ namespace CONFIG {
         RIGHT
     };
 
+
+    enum DrivingDirection {
+        SteerableWheelsAtFront,
```
[the remark this answers is not in the document]
Comment (robberwick) [reply libs/config/include/drivetrain_config.h]: `Car` and `Forklift` are improvements, but they're not `DrivingDirections`

```suggestion

    enum SteeringStyle {
        CAR = 1,
        FORKLIFT = -1
    };

```

**Review by robberwick — CHANGES_REQUESTED** [review]
close, but i'm a pedant. sorry.

**Review by robberwick — APPROVED** [review]
Excellent :+1:

**Pull request merged**